tgui/packages/tgui/interfaces/PlayerPreferences.tsx

```diff
@@ -0,0 +1,1502 @@
+      affectsSave = true;
+      break;
+    case 5:
+      CurrentTab = GameSettings;
+      break;
+    case 6:
+      CurrentTab = KeybindSetting;
+      break;
+    case 99:
+      CurrentTab = DEBUG_ENABLED ? DebugPanel : CurrentTab;
+      break;
+    default:
+  }
+
+  // I dont like this shit, but it doesn't matter in the end
+  // i'd rather massage the data in js than byond.
+  const slotNames = Object.values(save_slot_names).map(
+    name => name.split(' ')[0]
+  );
+
+  const saveSlots = new Array(10).fill(1).map((_, idx) => (
+    <Button
+      key={idx + 1}
+      selected={idx + 1 === slot}
+      onClick={() => act('changeslot', { changeslot: idx + 1 })}>
+      {slotNames[idx] || `Character ${idx + 1}`}
+    </Button>
+  ));
+
+  return (
+    <Window
+      width={1140}
+      height={650}>
+      <Window.Content scrollable>
+        <Flex>
+          <Flex.Item>
+            <NavigationSelector tabIndex={tabIndex} onTabChange={setTabIndex} />
+          </Flex.Item>
+          <Flex.Item grow={1} basis={0}>
+            {affectsSave ? (
+              <Section title="Save slot" buttons={saveSlots}>
+                <CurrentTab />
+              </Section>
+            ) : (
+              <CurrentTab />
+            )}
+          </Flex.Item>
+        </Flex>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const BackgroundInformation = (props, context) => {
+  const { act, data, config } = useBackend<PlayerPreferencesData>(context);
+  const {
+    slot,
+    flavor_text,
+    med_record,
+    gen_record,
+    sec_record,
+    exploit_record,
+  } = data;
+
+  const [characterDesc, setCharacterDesc] = useLocalState(
+    context,
+    'characterDesc' + slot,
+    flavor_text
+  );
+  const [medicalDesc, setMedicalDesc] = useLocalState(
+    context,
+    'medicalDesc' + slot,
+    med_record
+  );
+  const [employmentDesc, setEmploymentDesc] = useLocalState(
+    context,
+    'employmentDesc' + slot,
+    gen_record
+  );
+  const [securityDesc, setSecurityDesc] = useLocalState(
+    context,
+    'securityDesc' + slot,
+    sec_record
+  );
+  const [exploitsDesc, setExploitsDesc] = useLocalState(
+    context,
+    'exploitsDesc' + slot,
+    exploit_record
+  );
+  return (
+    <Section title="Background information">
+      <Section
+        title="Character Description"
+        buttons={
+          <Box>
+            <Button
+              icon="save"
+              disabled={characterDesc === flavor_text}
+              onClick={() => act('flavor_text', { characterDesc })}>
+              Save
+            </Button>
+            <Button
+              icon="times"
+              onClick={() => setCharacterDesc(flavor_text)}>
+              Reset
+            </Button>
+          </Box>
+        }>
+        <TextArea
+          key="character"
+          height="100px"
+          value={characterDesc}
+          onChange={(e, value) => setCharacterDesc(value)}
+        />
+      </Section>
+
+      <Grid>
```

Comment:
Grid is deprecated, replace with flex/stack or something else, or dont if you can't.

Reply:
Like i said, it's fine if you don't put grid with flex and stack.
Since there is only grid, it's okay